hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -404,12 +461,16 @@ public void saveToCleanRequested(HoodieInstant instant, Option<byte[]> content)
     // Write workload to auxiliary folder
     createFileInAuxiliaryFolder(instant, content);
     // Plan is only stored in auxiliary folder
-    createFileInMetaPath(instant.getFileName(), Option.empty());
+    createFileInMetaPath(instant.getFileName(), Option.empty(), false);
   }
 
-  private void createFileInMetaPath(String filename, Option<byte[]> content) {
+  private void createFileInMetaPath(String filename, Option<byte[]> content, boolean allowOverwrite) {
     Path fullPath = new Path(metaClient.getMetaPath(), filename);
-    createFileInPath(fullPath, content);
+    if (allowOverwrite || metaClient.getMetadataVersion().isNullVersion()) {
+      createFileInPath(fullPath, content);
+    } else {
+      writeFileOnceInPath(fullPath, content);
```

Comment:
why do we have to support the two modes of creating? seems the difference is just that one creates and then opens, while one open a new file , writes, closes?